### ISSUE TITLE
17.03.1-ce.... format doesn't follow http://semver.org/ causing breaks

### ISSUE THREAD

**Issue by schultz9999** [docker/for-mac#1491]
### Expected behavior
Version is to follow http://semver.org/ rules.

### Actual behavior
03 in minor doesn't follow the expected format required by semantic version.

```
  File "/usr/local/lib/python2.7/site-packages/dockerman/__init__.py", line 931, in docker_version
    return semantic_version.Version.coerce(c.version()['Version'], partial=True)
  File "/usr/local/lib/python2.7/site-packages/semantic_version/base.py", line 183, in coerce
    return cls(version, partial=partial)
  File "/usr/local/lib/python2.7/site-packages/semantic_version/base.py", line 77, in __init__
    major, minor, patch, prerelease, build = self.parse(version_string, partial)
  File "/usr/local/lib/python2.7/site-packages/semantic_version/base.py", line 212, in parse
    raise ValueError("Invalid leading zero in minor: %r" % version_string)
ValueError: Invalid leading zero in minor: u'17.03.1-ce'
```

### Information

  - Full output of the diagnostics from "Diagnose & Feedback" in the menu
  - A reproducible case if this is a bug, Dockerfiles FTW
  - Page URL if this is a docs issue or the name of a man page

### Steps to reproduce the behavior

Try to validate the version with Python's semantic_version library.

**Comment by avsm** [comment]
Thanks for the bug report.  This isn't directly a Docker for Mac bug, as the app follows the upstream Docker versioning conventions (with a `-mac` suffix to indicate its local epoch).

Docker doesn't currently use semantic versioning, but there is a proposal for switching to it in https://github.com/docker/docker/issues/31842. Once that discussion has concluded and is merged into Docker upstream, the Mac version here will follow. Please do register feedback on that issue, such as your desire for py-semantic_version compatibility.

**Comment by ijc25** [comment]
To be clear, https://github.com/docker/docker/issues/31842 is proposing the use of semver for the ReST API, which is independent from the docker version. AFAIK the docker versions have never been semver, lots of projects use 3 integers separated by a `.` without it being semver.

**Comment by schultz9999** [comment]
We didn't have any issues before this update. For example, I dug out 1.13.1 and things are good. Considering that it's very hard to download previous released and autoupdate is generally enabled, I can see this to break many. 

Speaking of older version, do you mind sharing a couple URLs for versions that have no '0' in MINOR?
